src/modules/reports/reports.controller.ts

```diff
@@ -0,0 +1,15 @@
+import { Controller, Get } from '@nestjs/common';
+import { ReportService } from './services/reports.service';
+import { IDashboardStats } from 'src/core/interfaces/dashboard-stats.interface';
+
+@Controller('report')
+export class ReportController {
+
+  constructor(private reportService: ReportService) {}
+
+  @Get('dashboard/stats')
+  async getDashboardStats(): Promise<any> {
```

Comment:
Instead of `Promise<any>` use `Promise<ICustomResponse>`. 

**Reason:**
We have pre defined strict type to return data from controller functions and that type is `ICustomResponse` so we have to use this type for all controller function with `Promise`.